### PR TITLE
Initial contribution of spec document

### PR DESCRIPTION
**Pull request by chkal**
A [CQ](http://dev.eclipse.org/ipzilla/show_bug.cgi?id=20980) for the specification document has been created a few days ago.

According to the [latest comment](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20980#c3), we can now proceed with the checkin.

This PR adds the specification document to a sub-directory `jaxrs-spec`. Please note that this PR contains the raw initial contribution WITHOUT any modifications. IMO we should get this one in quickly and then proceed with updating the structure/pom/etc according to our requirements afterwards.

I guess that everyone wants to get the spec document in as soon as possible. So I'm requesting to fast track it according to our project rules.

### CODE REVIEW

**Comment by chkal** [comment]
@kwsutter Thanks for your feedback. I guess you are right. I'll update the PR tomorrow.

**Comment by mkarg** [comment]
I'm not a big friend of putting the spec source *unchanged* into `master` as we have to "correct" it somehow before publication. Christian, maybe we could keep it in a *temporary feature branch* and work together on a transition into a form that is suitable as the first draft? I would be happy to help with that *before* we merge the PR. Too bad Github cannot turn PRs back into Drafts...

**Comment by kwsutter** [comment]
I would go ahead and merge the original source.  Much easier to do the PRs
against that.  We did it this way with the Platform specs and it worked out
well.

-- Kevin

**Comment by mkarg** [comment]
I cannot see the least what is easier in opening a PR against our master compared to opening a PR against christian's fork. But maybe this is because I'm used to Github and open source. ;-)

**Comment by mkarg** [comment]
@ivargrimstad @waynebeaton @eclipsewebmaster AFAIK the EF is very strict about IP, so it runs a bot which checks git authors against a database for some reason. Regarding the adoption of foreing contributions (like this one), is it OK if we (the committers) add our own name as the author (which obviously is a lie in most cases, but is the sole technical solution we have to convince the bot), or is there a trick to tell the bot to accept the original authors (which possibly have no Github account)?

**Comment by chkal** [comment]
I updated the PR and removed the `spec` folder which contained the original tex files.

My vote still goes for getting the initial contribution into our repository as soon as possible and continue the restructuring there. However, if anything would like to get some important first changes into this first version (like updates to the pom or something like that), feel free to open a pull request against the PR branch in my fork. This may be fine as long as we preserve the commit containing the unmodified initial contribution.

File: jaxrs-spec/pom.xml [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+        <asciidoctor.maven.plugin.version>2.0.0-RC.1</asciidoctor.maven.plugin.version>
+        <asciidoctorj.version>2.0.0</asciidoctorj.version>
+        <asciidoctorj.pdf.version>1.5.0-alpha.17</asciidoctorj.pdf.version>
+        <jruby.version>9.2.7.0</jruby.version>
+        <!-- status: DRAFT, BETA, etc., or blank for final -->
+        <status>DRAFT</status>
+    </properties>
+
+    <scm>
+        <connection>scm:git:git@github.com:eclipse-ee4j/restful-web-services</connection>
```
Comment (mkarg) [inline jaxrs-spec/pom.xml]: This URL produces a 404. I will send you a PR containing a valid URL.

Comment (chkal) [reply jaxrs-spec/pom.xml]: Good catch! Thanks!

Comment (mkarg) [reply jaxrs-spec/pom.xml]: Please review https://github.com/chkal/jaxrs-api/pull/3. :-)

Comment (chkal) [reply jaxrs-spec/pom.xml]: Merged

**Comment by chkal** [comment]
> LGTM, but maybe we should add the original authors to your initial commit using [`Co-authored-by`](https://help.github.com/en/articles/creating-a-commit-with-multiple-authors) or [`Also-By`](https://wiki.eclipse.org/Development_Resources/Handling_Git_Contributions#Multiple_Authors) tags?

I don't think that this is necessary. Other initial contributions were also committed simply by one of the project committers. Just my 2 cents.

**Comment by chkal** [comment]
@eclipse-ee4j/eclipse-jaxrs Please note that this PR now contains the original initial contribution and two minor fixes provided by @mkarg on top. See the inidividual commits for details.

**Comment by waynebeaton** [comment]
> is it OK if we (the committers) add our own name as the author

Yes. As you've already discovered, you can list other authors using Also-by.

It could be helpful (and avoid making the Author credentials a lie) if you describe the source of the content in the commit comment.

**Comment by chkal** [comment]
> Yes. As you've already discovered, you can list other authors using Also-by.

Sure, I could also list all the authors in `Also-by` headers, but this would require a full list which we don't have at the moment.

**Comment by mkarg** [comment]
> > Yes. As you've already discovered, you can list other authors using Also-by.
> 
> Sure, I could also list all the authors in `Also-by` headers, but this would require a full list which we don't have at the moment.

I do not think we need a full list of *all* authors. As 99.9% of the text was written solely by the *editors* (plus a few lines by Bill BURKE (@patriot1burke) and myself (@mkarg), it should be sufficient to list just the *editors*. According to the JCP, these had been:
* JSR 311 - JAX-RS 1.0 - Marc HADLEY (@hadleynet), Paul SANDOZ (@paulsandoz)
* JSR 311 - JAX-RS 1.1 - Marc HADLEY, Paul SANDOZ
* JSR 339 - JAX-RS 2.0 - Santiago PERICAS-GEERTSEN (@spericas), Marek POTOCIAR (@mpotociar)
* JSR 339 - JAX-RS 2.0 rev. A - Santiago PERICAS-GEERTSEN, Marek POTOCIAR
* JSR 370 - JAX-RS 2.1 - Pavel BUCEK (@pavelbucek), Santiago PERICAS-GEERTSEN

**Comment by waynebeaton** [comment]
> > Yes. As you've already discovered, you can list other authors using Also-by.
> 
> Sure, I could also list all the authors in `Also-by` headers, but this would require a full list which we don't have at the moment.

You don't need to (sorry, I should have made that more clear). The Eclipse Foundation has the information if anybody needs it. Capturing it in the Git commit is ideal, but is not absolutely required.

**Comment by chkal** [comment]
> You don't need to (sorry, I should have made that more clear). The Eclipse Foundation has the information if anybody needs it. Capturing it in the Git commit is ideal, but is not absolutely required.

Thanks for the clarification. Just a quick question. According to [this text](https://wiki.eclipse.org/Development_Resources/Handling_Git_Contributions#Multiple_Authors) authors added via `Also-by` need a CLA. Is this checked by the `eclipsefdn/eca` check? If so, the check will most likely fail if I add the previous authors.

**Comment by waynebeaton** [comment]
> @waynebeaton Thanks for the clarification. Just a quick question. According to [this text](https://wiki.eclipse.org/Development_Resources/Handling_Git_Contributions#Multiple_Authors) authors added via `Also-by` need a CLA. Is this checked by the `eclipsefdn/eca` check? If so, the check will most likely fail if I add the previous authors.

TL:DR: "Also-by" developers do not require an ECA.

The checker does not consider "Also-by" entries. FWIW, the process that harvests information to generate the charts that we put on project pages does look at Also-by entries (so you'll see "Also-by" authors as contributors o nthe charts).

That's an old page. I've taken a quick pass at updating it. I have an [issue](https://bugs.eclipse.org/bugs/show_bug.cgi?id=532678) open to update the handbook content with the useful bits on this page (and finally retire this page).

**Comment by chkal** [comment]
@waynebeaton Thanks a lot for the clarification.

@eclipse-ee4j/eclipse-jaxrs I updated the PR to list original authors/editors via `Also-by` entries in the commit message.

@kwsutter @mkarg @asoldano Could you resubmit your approval? Thanks!

**Comment by mkarg** [comment]
@eclipse-ee4j/eclipse-jaxrs Review please! :-)

**Comment by chkal** [comment]
It would be great if we could get one more approval, so we can merge this one and start with the work of cleaning up the document (see #804).

**Comment by mkarg** [comment]
@kwsutter I know you are pretty busy in Ludwigsburg, but maybe you find some time to cast your vote once again? :-)

**Comment by asoldano** [comment]
Noticed this late, but I would have re-approved ;-)

**Pull request merged**